src/cargo/ops/lockfile.rs

```diff
@@ -40,6 +64,8 @@ pub fn write_pkg_lockfile(ws: &Workspace, resolve: &Resolve) -> CargoResult<()>
         Ok(s)
     });
 
+    check_duplicate_pkgs(resolve).chain_err(|| format!("failed to generate lock file"))?;
```

Comment:
Interesting! I've found a place in resolve, where we do similar kind of verification:

https://github.com/rust-lang/cargo/blob/c51ed7b9f3595e6f9e742c778830ef279314443c/src/cargo/core/resolver/mod.rs#L142-L143

Let's move `check_duplicate_pkgs` call just below `check_cycles`?

Reply:
Thanks. This looks like the right place for the check.